lib/course_planner/csv_parser.ex

```diff
@@ -0,0 +1,36 @@
+defmodule CoursePlanner.CsvParser do
+  @moduledoc false
+
+  @csv_seperator ","
+
+  def parse(csv_data, column_count, trim_elements \\ true) do
+    splited_csv = csv_splitter(csv_data)
+    trimmed_splitted_csv =
+      if trim_elements do
+        trim_splitted_csv(splited_csv)
+      else
+        splited_csv
+      end
+
+    if validate_column_count(trimmed_splitted_csv, column_count) do
+      {:ok, Enum.chunk_every(trimmed_splitted_csv, column_count)}
+    else
+      {:error, "Input data is not matching the column number."}
+    end
+  end
+
+  defp trim_splitted_csv(splited_csv) do
+    Enum.map(splited_csv, &(String.trim(&1)))
+  end
+
+  defp csv_splitter(csv_data) do
+    String.split(csv_data, @csv_seperator)
+  end
+
+  defp validate_column_count(trimmed_splitted_csv, column_count) do
```

Comment:
Now this validation looks weird to me.
So you're ignoring the line-breaks and validating the sum of all record's columns?
I'm not sure if this is a good validation. I think every record should end in a line break and the validation should have a per row granularity.

Reply:
so what if I remove the new lines here and then validate?

Reply:
I think you should split first by line separator and validate them individually.
Let's say there's 10 lines but only one line is invalid. Should you insert the valid 9 records?
If not, you should specifically warn that line x is invalid. Think about inserting a massive CSV and then you get a generic "your CSV is invalid" warning.
This also removes the necessity to chunk the records and avoid false-positives: e.g. there's 5 lines of with 4 records each, it's gonna consider valid in terms of number of records if you don't validate the line break.

Reply:
I'll update the generic error too.
but I think it's better to insert only if everything is okay

Reply:
Done